### PR TITLE
nixos-rebuild-ng: improve IPv6 support

### PR DESCRIPTION
**Pull request by ungeskriptet**
These changes fix support for using an IPv6 address for the target host. This approach is similar to the one found in nix [1]. Previously, nixos-rebuild-ng would pass a host string to `ssh` that cannot be parsed.

Since the host string is being treated like a URL, we encode the scope ID symbol (%) as %25. This is required for compatibility with other nix commands (e.g. nix-copy-closure).

Example:
```
nixos-rebuild \
  --target-host "root@[fe80::9272:6aff:fe15:faf3%25enp4s0]" \
  --flake path:/etc/nixos#xiatian -L \
  switch
```

[1] https://github.com/NixOS/nix/blob/8b955d8/src/libstore/store-reference.cc#L143-L166

## Things done

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [x] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by binarymaster — APPROVED** [review]

File: pkgs/by-name/ni/nixos-rebuild-ng/src/nixos_rebuild/utils.py [hunk 1]
```diff
@@ -99,3 +100,22 @@ def format_row(row: Mapping[str, Any]) -> str:
...
+        host_fixed = host_split[-1].strip("[]").replace("%25", "%")
+        IPv6Address(host_fixed)
+        if len(host_split) > 1:
+          host = host_split[0] + "@" + host_fixed
+        else:
+          host = host_fixed
+    except AddressValueError:
+        host = host
+
+    return host
```
Comment (thiagokokada) [inline pkgs/by-name/ni/nixos-rebuild-ng/src/nixos_rebuild/utils.py]: Please add some tests for this function, otherwise I have zero idea what this is meant to do.

**Review by thiagokokada — CHANGES_REQUESTED** [review]
Please see the comments above.

File: pkgs/by-name/ni/nixos-rebuild-ng/src/nixos_rebuild/utils.py (hunk 1 above)
Comment (ungeskriptet) [reply pkgs/by-name/ni/nixos-rebuild-ng/src/nixos_rebuild/utils.py]: Thanks, fixed.

**Review by thiagokokada — APPROVED** [review]
LGTM, I would still prefer to have unit tests in the `ssh_host()` method instead of adding test cases for `process`, but otherwise the code looks good.

**Comment by thiagokokada** [comment]
Can you fix the imports?

**Review by thiagokokada — APPROVED** [review]
